api/geth_backend.go

```diff
@@ -783,6 +783,14 @@ func (b *GethStatusBackend) AppStateChange(state string) {
 	b.log.Info("App State changed", "new-state", s)
 	b.appState = s
 
+	wallet, err := b.statusNode.WalletService()
+	if err != nil {
+		log.Error("Can't get wallet service", "error", err)
+		return
+	}
+
+	wallet.SetAppActive(s == appStateForeground)
```

Comment:
I have a different approach -- couldn't we start and stop wallet service when the app state changes? That would prevent from spreading `Service` between all structs in `services/wallet`.

Reply:
I'm just not sure how to do this so that it properly recover after restart

Reply:
let me try

Reply:
It depends how well `wallet.Service` is built. In theory, it has `Start()` and `Stop()` so it should work out of the box:

```go
if (s == appStateForeground) {
  err := wallet.Start(b.GethNode().Server())
  // check for err
  wallet.StartReactor(...) // this is missing from Start and it's fine if it needs to be separate
} else if (s == appStateBackground) {
  wallet.Stop()
}
```

But we need an integration test (in `service_test.go`) to verify it. Immediate thing I noticed is that `Start` does not do `StartReactor` so that would need to be called manually.

Reply:
@adambabik does it look better now?